### PR TITLE
Add `beforeError` hook

### LINKED ISSUE #412: `beforeError` hook
Like Got: https://github.com/sindresorhus/got/blob/main/documentation/9-hooks.md#beforeerror

### PR DESCRIPTION
**Pull request by uchihamalolan**
Fixes #412

### CODE REVIEW

**Comment by sindresorhus** [comment]
Linting is not passing. Make sure you always run `npm test` locally before pushing commits.

**Comment by sindresorhus** [comment]
It needs to be documented in the readme too.

**Comment by uchihamalolan** [comment]
@sindresorhus I see this error which is not related to my changes:
`Error: /Users/runner/work/ky/ky/test/helpers/with-page.ts: line 2, col 29, Error - 'ava/use-test' rule is disabled but never reported. (eslint-comments/no-unused-disable)`

any idea what to be done about it?

**Comment by uchihamalolan** [comment]
@sindresorhus Any update in this?

File: readme.md [hunk 1]
```diff
@@ -258,6 +258,33 @@ const response = await ky('https://example.com', {
 });
 ```
 
+###### hooks.beforeError
+
+Type: `Function[]`\
+Default: `[]`
+
+This hook enables you to modify the HTTPError right before it is thrown. The hook function receives `HTTPError` as argument and should return instance of HTTPError.
```
Comment (sindresorhus) [inline readme.md:266]:
```suggestion
This hook enables you to modify the `HTTPError` right before it is thrown. The hook function receives `HTTPError` as argument and should return instance of `HTTPError`.
```

File: readme.md [hunk 2]
```diff
@@ -258,6 +258,33 @@ const response = await ky('https://example.com', {
...
+
+```js
+import ky from 'ky';
+
+await ky('https://example.com', {
+	hooks: {
+		beforeError: [
+			(error) => {
```
Comment (sindresorhus) [inline readme.md:274]:
```suggestion
			error => {
```

**Pull request merged**